bridge-token-factory/src/lib.rs

```diff
@@ -342,6 +413,145 @@ impl BridgeTokenFactory {
             );
     }
 
+    pub fn get_deposit_token_fee(&self, token: &String) -> Option<DepositFee> {
+        self.deposit_fee.get(token)
+    }
+
+    pub fn get_withdraw_token_fee(&self, token: &String) -> Option<WithdrawFee> {
+        self.withdraw_fee.get(token)
+    }
+
+    // Returns fee percentage for different tokens per silo
+    pub fn get_withdraw_fee_percentage_per_silo(
+        &self,
+        silo_account_id: &AccountId,
+        token: &String,
+    ) -> Option<U128> {
+        self.withdraw_fee_percentage_per_silo
+            .get(&get_silo_fee_map_key(&silo_account_id, Some(token)))
+            .or_else(|| {
+                self.withdraw_fee_percentage_per_silo
+                    .get(&get_silo_fee_map_key(&silo_account_id, None))
+            })
+    }
+
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_deposit_fee(
+        &mut self,
+        token: &String,
+        eth_to_near: U128,
+        eth_to_aurora: U128,
+        lower_bound: U128,
+        upper_bound: U128,
+    ) {
+        let _ = validate_eth_address(token.clone());
+        self.deposit_fee.insert(
+            token,
+            &DepositFee {
+                fee_percentage: DepositFeePercentage {
+                    eth_to_near,
+                    eth_to_aurora,
+                },
+                bounds: FeeBounds {
+                    lower_bound,
+                    upper_bound,
+                },
+            },
+        );
+    }
+
+    //Fee should be added as per: 10% -> 0.1 = 0.1*10^6 with proper fee amount bounds
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee(
+        &mut self,
+        token: &String,
+        near_to_eth: U128,
+        aurora_to_eth: U128,
+        lower_bound: U128,
+        upper_bound: U128,
+    ) {
+        let _ = validate_eth_address(token.clone());
+        self.withdraw_fee.insert(
+            token,
+            &WithdrawFee {
+                fee_percentage: WithdrawFeePercentage {
+                    near_to_eth,
+                    aurora_to_eth,
+                },
+                bounds: FeeBounds {
+                    lower_bound,
+                    upper_bound,
+                },
+            },
+        );
+    }
+
+    /*
+    Sets fee percentage for withdraw per silo per token
+    @param: fee_percent:- it should be with 6 decimals, eg: 10% = 0.1*10^6
+            silo_account_id:- should be an aurora account
+            token:- token address, if passed `None` than default value is set.
+     */
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee_percentage_for_token_per_silo(
+        &mut self,
+        silo_account_id: AccountId,
+        token: Option<String>,
```

Comment:
We need to convert the token address to lowercase, otherwise, it will be possible to pass the valid uppercase address